File: app/src/main/java/com/example/optoapp/domain/model/Paciente.kt
package com.example.optoapp.domain.model

import java.time.LocalDate

data class PacienteModel(
    val id: String,
    val nombreCompleto: String,
    val edad: Int,
    val telefono: String,
    val fechaCreacion: LocalDate,
    val dni: String? = null,
    val fechaNacimiento: LocalDate? = null,
    val sexo: String? = null,
    val email: String? = null,
    val historiaOptometrica: String? = null,
    val direccion: String? = null,
    val distrito: String? = null,
    val ocupacion: String? = null,
    val acompanante: String? = null,
    val hobbies: String? = null,
    val ultimasEtiquetas: List<String> = emptyList(),
    val opticaId: String
)

File: web/src/domain/models/Paciente.ts
import { z } from "zod";

export const PacienteSchema = z.object({
  id: z.string(),
  nombre_completo: z.string(),
  edad: z.number(),
  telefono: z.string(),
  fecha_creacion: z.string(),
  dni: z.string().nullable().optional(),
  fecha_nacimiento: z.string().nullable().optional(),
  sexo: z.string().nullable().optional(),
  email: z.string().nullable().optional(),
  historia_optometrica: z.string().nullable().optional(),
  direccion: z.string().nullable().optional(),
  distrito: z.string().nullable().optional(),
  ocupacion: z.string().nullable().optional(),
  acompanante: z.string().nullable().optional(),
  hobbies: z.string().nullable().optional(),
  ultimas_etiquetas: z.string().nullable().optional(),
  optica_id: z.string(),
});

export type Paciente = z.infer<typeof PacienteSchema>;
